app/scripts/pages/dashboard.jsx

```diff
@@ -10,17 +10,64 @@ import FolderOptions from '../components/folderComponents/folderOptions.jsx';
 import ProjectOptions from '../components/projectComponents/projectOptions.jsx';
 import TagManager from '../components/globalComponents/tagManager.jsx'
 import VersionUpload from '../components/fileComponents/versionUpload.jsx';
+import { Path } from '../util/urlEnum';
 
 @observer
 class Dashboard extends React.Component {
     componentDidMount() {
         if (mainStore.projects.length === 0) {
             mainStore.getProjects(null, null);
         };
+        dashboardStore.setRouter(this.props.router);
+
         if(mainStore.openTagManager) mainStore.toggleTagManager();
+        mainStore.leftMenuDrawer.get('open') ? mainStore.toggleLeftMenuDrawer() : null;
+        this.loadView();
+    }
+
+    componentDidUpdate(prevProps) {
+        if(prevProps.params.id !== this.props.params.id) {
```

Comment:
If you've navigated away and back then do you need to also `dashboardStore.setRouter(this.props.router);`?

Reply:
Yea, i removed setRouter, it was only ever intended as a patch and was no longer useful